README.md

```diff
@@ -1,3 +1,43 @@
 # Zed Nu
 
 This extension adds support for the [Nu](https://github.com/nushell/nushell) language.
+
+## Override Default Configuration (Optional)
+
+You can minimize the configuration autoloaded when the server starts,
+which may improve performance as complicated Nushell configurations can slow down the language server.
+
+```json
+{
+  "lsp": {
+    "nu": {
+      "binary": {
+        "command": "nu",
+        "arguments": ["--config", "~/.config/nushell/lsp.nu", "--lsp"]
+      }
+    }
+  }
+}
+```
+
+### Example of Minimal lsp.nu
+
+```nushell
+# Configure PATH to search for external command completions
+$env.PATH = $env.PATH
+| split row (char esep)
+| append ($env.HOME | path join ".cargo" "bin")
+| uniq
+
+# Set up external completer (requires carapace)
```

Comment:
why does the lsp need an external completer?

Reply:
Basically works as a database for external arguments/subcommands completion.

Reply:
For example, this comes from carapace

<img width="392" alt="image" src="https://github.com/user-attachments/assets/f8895307-6a60-4b45-8a91-5771348a9b18" />

Of course, a collection of `extern`s of external commands may result in similar results. But delegating to `carapace` makes the engine_state for lsp smaller in memory, which is the way I prefer.